src/nwp_consumer/internal/inputs/ecmwf/_models.py

```diff
@@ -56,9 +56,15 @@ def filepath(self) -> str:
         return f"ecmwf/{self.fname}"
 
     def it(self) -> dt.datetime:
-        """Overrides the corresponding method in the parent class."""
-        return dt.datetime.strptime(self.fname[3:10], "%m%d%H%M").replace(
-            year=dt.datetime.now(tz=dt.UTC).year, tzinfo=dt.UTC,
+        """Overrides the corresponding method in the parent class.
+
+        The file name doesn't have the year in it, so we've added it.
+        This might be a problem around the new year.
+        """
+        return dt.datetime.strptime(
```

Comment:
What's the difference between this and the previous method of year replacement?

Reply:
First bit of code fails on a leap day as 1900 doesn't have a leap day

Can you merge, deploy etc?